metals/src/main/scala/scala/meta/internal/metals/clients/language/MetalsLanguageClient.scala

```diff
@@ -65,9 +82,22 @@ trait MetalsLanguageClient
    *         the input box should be dismissed in the editor.
    */
   @JsonRequest("metals/quickPick")
-  def metalsQuickPick(
+  private[clients] def rawMetalsQuickPick(
+      params: MetalsQuickPickParams
+  ): CompletableFuture[RawMetalsQuickPickResult]
+
+  final def metalsQuickPick(
       params: MetalsQuickPickParams
-  ): CompletableFuture[Option[MetalsQuickPickResult]]
+  )(implicit
+      ec: ExecutionContext
+  ): CompletableFuture[Option[MetalsQuickPickResult]] =
+    rawMetalsQuickPick(params).asScala.map { result =>
+      if (result.cancelled != null && result.cancelled) {
+        None
+      } else {
+        Option(result.itemId).map(MetalsQuickPickResult(_))
+      }
+    }.asJava
```

Comment:
Nitpick to conversions: java future -> scala future -> java future

Maybe replace `.asScala.map` with `thenApply` ?

Reply:
With `thenApply` execution context won't be needed, good point!